\begin{center}
    \Large\bfseries Exploring Structural Variation in Tumor Evolution through 
    Nanopore Sequencing

\end{center} 

\vspace{0.5cm}
    \noindent\textbf{\large Abstract:}

    Structural variants (SVs) are genomic alterations encompassing deletions, 
    insertions, and segment rearrangements, ranging from kilobases to entire 
    chromosomes. Despite their significance as biomarkers in oncological 
    diseases, these variants have remained relatively unexplored compared to 
    single nucleotide variants, largely due to the inherent limitations of 
    short-read sequencing technologies that have dominated large-scale genome 
    sequencing projects. This scenario has undergone a transformative change 
    with the advent of long-read sequencing technologies, which have enabled 
    the achievement of the first truly complete human telomere-to-telomere 
    reference genome, successfully filling gaps that short reads could not 
    resolve. This project focuses on conducting a comprehensive performance 
    evaluation of long-read-based structural variant callers, specifically in 
    the context of tumor evolution analysis. To address the limited availability
    of appropriate datasets, we have developed specialized workflows leveraging 
    high-performance computing resources for generating synthetic data with 
    custom SVs, thus facilitating robust benchmarking of various structural 
    variant detection methods. This computational approach enables systematic 
    evaluation of SV detection algorithms under controlled conditions, 
    providing valuable insights into their performance and reliability.


\vspace{0.5cm}
    \noindent\textbf{\large Key words:} 
    
    Structural Variants, Cancer Genomics, Oxford Nanopore Sequencing, Long-Read 
    Sequencing, High Performance Computing, Synthetic Data Generation, Variant 
    Calling, Bioinformatics.

